source/xep/httpFileUpload/DownloadManager.cpp

```diff
@@ -175,18 +127,14 @@ void DownloadManager::downloadFinished(QNetworkReply *reply)
     }
     else
     {        
-        QString filename = saveFileName(reply->property(kOriginalUrl).toString());
-
-
-        if (saveToDisk(filename, reply, reply->property(kIvAndKey).toString()))
-        {
-            printf("Download of %s succeeded (saved to %s)\n",
-                   url.toEncoded().constData(), qPrintable(filename));
-
-            emit httpDownloadFinished(filename);
-        }
+        file->write(reply->readAll());
+        file->close();
+        emit httpDownloadFinished(file->fileName());
     }
 
+    file->deleteLater();
+
+    downloadedFiles.remove(reply);
```

Comment:
On line 89 you asked the class DownloadManager to handle the resource of FileWithCypher for you. You never free it by your own. This is probably not what you want. The memory for each new download is growing until one closes the app. Only then, all instances of FileWithCypher gets destroyed.